### PR TITLE
Reintroduce name sections PR

### PR DESCRIPTION
**Pull request by NikVolf**
previous is #132 

Resolves https://github.com/paritytech/parity-wasm/issues/129

### CODE REVIEW

File: src/elements/name_section.rs [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+        serialize_test(original.clone());
+    }
+
+    #[test]
+    fn handles_real_wasm_file() {
+        let module = deserialize_file("./res/cases/v1/names.wasm").expect("Should be deserialized");
+
+        // Find the custom section containing function names.
+        //
+        // TODO: Add a real API for this.
```
Comment (pepyakin) [inline src/elements/name_section.rs]: Don't we have a real API for this? What about `.parse_names()`?

File: src/elements/name_section.rs [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+            }
+        }
+        let function_names = function_names_opt.expect("no name section found");
+
+        assert_eq!(
+            function_names.names().get(0).expect("name 0 missing"),
+            "double"
+        );
+
+        // TODO: Find a test file with module and local names.
```
Comment (pepyakin) [inline src/elements/name_section.rs]: I think this TODO is solved now, isn't it?

**Comment by NikVolf** [comment]
@pepyakin addressed!

**Comment by pepyakin** [comment]
also thanks to @emk for starting it!

**Comment by emk** [comment]
@pepyakin @NikVolf Thank you for handling the integration between my code and the main API! I got caught up with other issues (at work and in life), and I'm glad you were able to get this shipped. Thank you!

**Pull request merged**